### DISCUSSION TITLE
fatal error: postgres.h: No such file or directory

### DISCUSSION THREAD

**Discussion by Vineet-Sharma29** [osm-search/Nominatim#2223] [Q&A] [answered]
I had installed `nominatim` using docker(as given [here](https://github.com/mediagis/nominatim-docker)). However, in order to make changes in current codebase I am trying to install and run code in `master` branch as given [here](https://nominatim.org/release-docs/develop/appendix/Install-on-Ubuntu-18/). On running make command I am getting, following error:-

`fatal error: postgres.h: No such file or directory`.

Complete error log as follows:-

```
/srv/nominatim/Nominatim/module/Makefile:11: warning: overriding recipe for target 'install'
/usr/lib/postgresql/13/lib/pgxs/src/makefiles/pgxs.mk:241: warning: ignoring old recipe for target 'install'
/srv/nominatim/Nominatim/module/nominatim.c:1:10: fatal error: postgres.h: No such file or directory
 #include "postgres.h"
          ^~~~~~~~~~~~
compilation terminated.
<builtin>: recipe for target 'nominatim.o' failed
make[3]: *** [nominatim.o] Error 1
module/CMakeFiles/nominatim_lib.dir/build.make:60: recipe for target '/srv/nominatim/Nominatim/module/dummy' failed
make[2]: *** [/srv/nominatim/Nominatim/module/dummy] Error 2
CMakeFiles/Makefile2:1114: recipe for target 'module/CMakeFiles/nominatim_lib.dir/all' failed
make[1]: *** [module/CMakeFiles/nominatim_lib.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

I have tried to fix it my own by running `sudo apt install postgresql-server-dev-10` as suggested in some stackoverflow answers, however it did not work.

**Comment by mtmail** [comment, marked as answer]
https://github.com/mediagis/nominatim-docker is a separate project, they don't support Nominatim beyond the stable release 3.6 yet. Too many parts of the installation changed, one probably has to change the Dockerfile a lot to make the master branch work.

**Reply by Vineet-Sharma29** [reply]
I am not using `docker` image anymore and trying to install code in `master` branch as per instructions given [here](https://nominatim.org/release-docs/develop/appendix/Install-on-Ubuntu-18/), which is giving me the error.

**Reply by mtmail** [reply]
> /usr/lib/postgresql/13/lib/pgxs/src/makefiles/pgxs.mk:241: warning: ignoring old recipe for target 'install'

Cmake is looking for a `postgresql.h` for postgresql version 13. Do you have postgresql-server-dev-13 installed? Do you see postgresql.h for that version anywhere on the file system? If you have both 12 and 13 installed it might be easier to fully remove 13 first.

**Reply by Vineet-Sharma29** [reply]
Thanks!!! That solved the error. I uninstalled all versions other than `10` using instructions given [here](https://askubuntu.com/questions/32730/how-to-remove-postgres-from-my-installation).(care needs to taken while deletion as data is also deleted; in my case I had sql dump of data)
